lib/Transforms/Scalar/Scalarizer.cpp

```diff
@@ -290,6 +292,10 @@ bool Scalarizer::doInitialization(Module &M) {
 }
 
 bool Scalarizer::runOnFunction(Function &F) {
+  if (F.getParent()->HasDxilModule())
+    if (F.getParent()->GetDxilModule().GetShaderModel()->IsSM69Plus())
+      return false;
```

Comment:
Do we actually want to turn off scalarization entirely?  Maybe this is one place where we need to preserve it for vec1 only?

Reply:
Proper support for vec1 was missing from this change. I've added it now.